relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -661,12 +659,91 @@ mod tests {
         )
         .unwrap();
         let mut metrics = vec![];
-        extract_transaction_metrics(&config, None, event.value().unwrap(), &mut metrics);
+        extract_transaction_metrics(&config, None, &[], event.value().unwrap(), &mut metrics);
         assert_eq!(metrics.len(), 1);
 
         for metric in metrics {
             assert_eq!(metric.tags.len(), 1);
             assert!(!metric.tags.contains_key("satisfaction"));
         }
     }
+
+    #[test]
+    fn test_conditional_tagging() {
+        let json = r#"
+        {
+            "type": "transaction",
+            "transaction": "foo",
+            "start_timestamp": "2021-04-26T08:00:00+0100",
+            "timestamp": "2021-04-26T08:00:02+0100",
+            "measurements": {
+                "lcp": {"value": 41}
+            }
+        }
+        "#;
+
+        let event = Annotated::from_json(json).unwrap();
+
+        let config: TransactionMetricsConfig = serde_json::from_str(
+            r#"
+        {
+            "extractMetrics": [
+                "d:transactions/duration@millisecond"
+            ]
+        }
+        "#,
+        )
+        .unwrap();
+
+        let tagging_config: Vec<TaggingRule> = serde_json::from_str(
+            r#"
+        [
+            {
+                "condition": {"op": "gte", "name": "transaction.duration", "value": 9001},
+                "targetMetrics": ["d:transactions/duration@millisecond"],
+                "targetTag": "satisfaction",
+                "tagValue": "frustrated"
+            },
+            {
+                "condition": {"op": "gte", "name": "transaction.duration", "value": 666},
+                "targetMetrics": ["d:transactions/duration@millisecond"],
+                "targetTag": "satisfaction",
+                "tagValue": "tolerated"
+            },
+            {
+                "condition": {"op": "and", "inner": []},
+                "targetMetrics": ["d:transactions/duration@millisecond"],
+                "targetTag": "satisfaction",
+                "tagValue": "satisfied"
+            }
+        ]
+        "#,
+        )
+        .unwrap();
+
+        let mut metrics = vec![];
+        extract_transaction_metrics(
+            &config,
+            None,
+            &tagging_config,
+            event.value().unwrap(),
+            &mut metrics,
+        );
+        assert_eq!(
+            metrics,
+            &[Metric::new_mri(
+                METRIC_NAMESPACE,
+                "duration",
+                MetricUnit::Duration(DurationUnit::MilliSecond),
+                MetricValue::Distribution(2000.0),
+                UnixTimestamp::from_secs(1619420402),
+                {
+                    let mut tags = BTreeMap::new();
+                    tags.insert("satisfaction".to_owned(), "tolerated".to_owned());
+                    tags.insert("transaction".to_owned(), "foo".to_owned());
+                    tags
+                }
+            )]
+        );
+    }
```

Comment:
Should we also test a conditional that has a different field, e.g. `measurements.lcp`?

Reply:
yup will add